src/createTemplate.js

```diff
@@ -2,15 +2,37 @@ const { Template } = require('webpack');
 const { NAMESPACE } = require('./constants');
 
 const afterModule = `
-if (module.hot) {
+const exports = module.exports || module.__proto__.exports;
+let shouldBind = false;
+
+if (!exports || typeof exports != 'object') {
+  shouldBind = false;
+} else {
+  for (const key in exports) {
+    if (key === '__esmodule') continue;
+    const exportValue = exports[key];
+    if (typeof exportValue == 'function') {
+      const name = exportValue.name || exportValue.displayName;
+      if (name) {
+        shouldBind = shouldBind || (typeof name === 'string' && name[0] == name[0].toUpperCase());
```

Comment:
A function starting with a capital is considered a component

Reply:
Another check that could work here would be to test for the stringified function body containing h(), createElement() or html`:

```js
if (/(h\(|createElement\(|html`)/.test(exportValue)) {
  shouldBind = true;
}
```